src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1221 @@
+#include <vector>
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <iomanip>
+
+
+namespace {
+
+const uint32_t maxDebugIpType = TRACE_S2MM_FULL+1;
+
+static const char* debugIpNames[maxDebugIpType] = {
+  "unknown",
+  "LAPC",
+  "ILA",
+  "AIM",
+  "TraceFunnel",
+  "TraceFifoLite",
+  "TraceFifoFull",
+  "AM",
+  "ASM",
+  "AxiStreamProtocolChecker",
+  "TS2MM",
+  "AxiDMA",
+  "TS2MMFull"
+};
+
+
+class DebugIpStatusCollector
+{
+
+  xclDeviceHandle handle;
+
+  std::vector<char> map;
+
+  uint64_t debugIpNum[maxDebugIpType];
+  bool     debugIpOpt[maxDebugIpType];
+
+  size_t   cuNameMaxStrLen[maxDebugIpType];
+  size_t   portNameMaxStrLen[maxDebugIpType];
+
+  std::vector<std::string> cuNames[maxDebugIpType];
+  std::vector<std::string> portNames[maxDebugIpType];
+
+  xclDebugCountersResults          aimResults;
+  xclStreamingDebugCountersResults asmResults;
+  xclAccelMonitorCounterResults    amResults;
+  xclDebugCheckersResults          lapcResults;
+  xclDebugStreamingCheckersResults spcResults;
+
+public :
+  DebugIpStatusCollector(xclDeviceHandle h);
+  ~DebugIpStatusCollector() {}
+
+  void collect();
+  void collect(const std::vector<std::string> & _elementsFilter);
+
+  void printOverview(std::ostream& _output);
+  void printAllResults(std::ostream& _output);
+
+  void populateOverview(boost::property_tree::ptree &_pt);
+  void populateAllResults(boost::property_tree::ptree &_pt);
+
+private :
+
+  debug_ip_layout* getDebugIpLayout();
+
+  void getDebugIpData();
+  void getCuNamePortName(uint8_t type, std::string& dbgIpName, std::string& cuName, std::string& portName); 
+  void getStreamName(uint8_t type, std::string& dbgIpName, std::string& masterName, std::string& slaveName); 
+
+  void readAIMCounter(debug_ip_data*);
+  void readAMCounter(debug_ip_data*);
+  void readASMCounter(debug_ip_data*);
+  void readLAPChecker(debug_ip_data*);
+  void readSPChecker(debug_ip_data*);
+
+  void printAIMResults(std::ostream&);
+  void printAMResults(std::ostream&);
+  void printASMResults(std::ostream&);
+  void printLAPCResults(std::ostream&);
+  void printSPCResults(std::ostream&);
+
+  void populateAIMResults(boost::property_tree::ptree &_pt);
+  void populateAMResults(boost::property_tree::ptree &_pt);
+  void populateASMResults(boost::property_tree::ptree &_pt);
+  void populateLAPCResults(boost::property_tree::ptree &_pt);
+  void populateSPCResults(boost::property_tree::ptree &_pt);
+
+  void processElementFilter(const std::vector<std::string> & _elementsFilter);
+};
+
+
+
+DebugIpStatusCollector::DebugIpStatusCollector(xclDeviceHandle h)
+    : handle(h)
+    , debugIpNum{0}
+    , debugIpOpt{false}
+    , cuNameMaxStrLen{0}
+    , portNameMaxStrLen{0}
+    , aimResults{0}
+    , asmResults{0}
+    , amResults{0}
+    , lapcResults{0}
+    , spcResults{0}
+{
+  // By default, enable status collection for all Debug IP types
+  for(uint64_t i = 0; i < maxDebugIpType ; ++i) {
+    debugIpOpt[i] = true;
+  }
```

Comment:
why not initialize in the constructor initializer list?

Reply:
If constructor initializer list is used, then only the first element gets initialized to "true" and rest of the elements remain "false". I had tried this earlier and because of the issue had to set all the elements to "true" in the current manner.

Reply:
ok. you are right indeed.   then how about 
`std::fill(debugIpOpt, debugIpOpt + maxDebugIpType, true);`

Reply:
That works. I have added the change.